### ISSUE TITLE
[no-unused-vars] v4 false positive: in d.ts files

### ISSUE THREAD

**Issue by doberkofler** [typescript-eslint/typescript-eslint#2456]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JS
module.exports = {
	parser: '@typescript-eslint/parser',
	parserOptions: {
		project: './tsconfig.json'
	},
	extends: [
		'eslint:recommended',
		'plugin:@typescript-eslint/recommended',
		'plugin:@typescript-eslint/recommended-requiring-type-checking',
	],
	rules: {
		'@typescript-eslint/explicit-function-return-type': 'off',
		'@typescript-eslint/indent': ['warn', 'tab'],
	}
};
```

**foo.d.ts**:
```TS
interface Foo {
	foo: boolean,
}
```

```JSON
{
  "compilerOptions": {
    "allowJs": true,
    "checkJs": false,
    "outDir": "./dist",
    "outFile": "./dist/index.js",
    "moduleResolution": "node",
    "target": "es5",
    "module": "system",
    "sourceMap": true,
    "baseUrl": "./"
  },
  "compileOnSave": false,
  "include": [
    "*"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

**Expected Result**

in a `d.ts` file all interfaces are implicitly exported and therefore 'used'

**Actual Result**

  1:11  warning  'Foo' is defined but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (0 errors, 1 warning)

**Additional Info**

no

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.0.1` |
| `@typescript-eslint/parser`        | `4.0.1` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.8.0` |
| `node`                             | `14.8.0` |
